compiler/rustc_typeck/src/check/callee.rs

```diff
@@ -374,7 +374,28 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                                     |p| format!("`{}` defined here returns `{}`", p, callee_ty),
                                 )
                             }
-                            _ => Some(format!("`{}` defined here", callee_ty)),
+                            _ => {
+                                match def {
+                                    // Emit a different diagnostic for local variables, as they are not
+                                    // type definitions themselves, but rather variables *of* that type.
+                                    Res::Local(hir_id) => Some(format!(
+                                        "`{}` has type `{}`",
+                                        self.tcx.hir().name(hir_id),
+                                        callee_ty
+                                    )),
+                                    Res::Def(kind, def_id) => {
+                                        if let Some(Namespace::ValueNS) = kind.ns() {
+                                            Some(format!(
+                                                "`{}` defined here",
+                                                self.tcx.def_path_str(def_id),
+                                            ))
+                                        } else {
+                                            Some(format!("`{}` defined here", callee_ty))
+                                        }
+                                    }
```

Comment:
```suggestion
                                    Res::Def(kind, def_id) if let Some(Namespace::ValueNS) = kind.ns() => {
                                        Some(format!(
                                            "`{}` defined here",
                                            self.tcx.def_path_str(def_id),
                                        ))
                                    }
```
Since this is just a one time thing we can probably simplify it with an if let match guard.

Reply:
Why was this resolved?

Reply:
@pickfire Do you know why this doesn't work?